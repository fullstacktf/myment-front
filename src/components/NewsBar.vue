<template> <!-- Just a cool way to show our news -->
    <div class="containerBar marquee" >
        <div >{{newNews}}</div>
    </div>
</template>

<script>
export default {
    props: {
        newNews: {
            type: String,
            required: true,
            default: "if you are watching it, some error may has happened or its just a random text for test it. Use the 'newNews' prop for change me. We will show in this bar some useful stuff"
        }

    }

}
</script>

<style>

.marquee {
    display:flex;
    align-items: center;
    font-size: 25px;
    width: 100%;
    height: 50px;
	background-color: rgb(0, 0, 0);
	color: white;
    white-space: nowrap;
    overflow: hidden;
    box-sizing: border-box;
    box-shadow: inset 3px 3px 4px white, inset -3px -3px 4px white;
}
.marquee div {
    display: inline-block;
    padding-left: 100%;
    animation: marquee 15s linear infinite;
}
@keyframes marquee {
    0%   { transform: translate(0, 0); }
    100% { transform: translate(-100%, 0); }
}

</style>
