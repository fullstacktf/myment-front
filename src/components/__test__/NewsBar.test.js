import NewsBar from '../NewsBar.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Buefy from 'buefy';
import mock from '../../test/mockBack';

describe('NewsBar', () => {
  let localVue;

  beforeEach(() => {
    const Vue = createLocalVue();
    Vue.use(Buefy);
    localVue = Vue;
  });
  test('render component', () => {
    const wrapper = shallowMount(NewsBar, {
      localVue,
      propsData: {
        newNews: "Hi I'm news bar letter",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Check props', () => {});

  test('check Contents props', () => {});
});
