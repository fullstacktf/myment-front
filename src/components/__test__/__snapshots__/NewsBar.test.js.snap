// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsBar render component 1`] = `"<div class=\\"containerBar marquee\\"><div>Hi I'm news bar letter</div></div>"`;
